Sources/BraveWallet/Extensions/WebImageDownloaderExtensions.swift

```diff
@@ -0,0 +1,25 @@
+// Copyright 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import BraveCore
+import BraveUI
+
+extension BraveCore.WebImageDownloader: WebImageDownloaderType {
+  @MainActor public func downloadImage(url: URL) async -> (UIImage?, Error?) {
```

Comment:
Do these methods need to be main actor isolated?

Reply:
it will give a crash (EXC_BREAKPOINT) once you start loading NFT images without the main actor isolated

Reply:
Okay fine to keep the main actor isolation for brave core's, most likely some sort of threading issue on the chromium side

Reply:
can confirm no crash using SDWebImage without being main actor isolated to load NFT images. so yeah must be some threading issues on the chromium side.